x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -141,172 +102,77 @@ const AssistantComponent: React.FC<Props> = ({
-          updatedConversation = await getConversation(conversationId);
-        }
-
-        if (updatedConversation) {
-          setCurrentConversation(updatedConversation);
-        }
-
-        return updatedConversation;
-      }
-    },
-    [conversations, currentConversation?.id, getConversation]
-  );
-
-  useEffect(() => {
-    if (areConnectorsFetched && conversationsLoaded && Object.keys(conversations).length > 0) {
-      setCurrentConversation((prev) => {
-        const nextConversation =
-          (currentConversationId && conversations[currentConversationId]) ||
-          (isAssistantEnabled &&
-            (conversations[getLastConversationId(conversationTitle)] ||
-              find(conversations, ['title', getLastConversationId(conversationTitle)]))) ||
-          find(conversations, ['title', getLastConversationId(WELCOME_CONVERSATION_TITLE)]);
-
-        if (deepEqual(prev, nextConversation)) return prev;
-
-        const conversationToReturn =
-          (nextConversation &&
-            conversations[
-              nextConversation?.id !== '' ? nextConversation?.id : nextConversation?.title
-            ]) ??
-          conversations[WELCOME_CONVERSATION_TITLE] ??
-          getDefaultConversation({ cTitle: WELCOME_CONVERSATION_TITLE });
-
-        // updated selected system prompt
-        setEditingSystemPromptId(
-          getDefaultSystemPrompt({
-            allSystemPrompts,
-            conversation: conversationToReturn,
-          })?.id
-        );
-        if (
-          prev &&
-          prev.id === conversationToReturn.id &&
-          // if the conversation id has not changed and the previous conversation has more messages
-          // it is because the local conversation has a readable stream running
-          // and it has not yet been persisted to the stored conversation
-          prev.messages.length > conversationToReturn.messages.length
-        ) {
-          return {
-            ...conversationToReturn,
-            messages: prev.messages,
-          };
-        }
-        return conversationToReturn;
-      });
-    }
-  }, [
+  const {
+    currentConversation,
+    currentSystemPromptId,
+    handleCreateConversation,
+    handleOnConversationDeleted,
+    handleOnConversationSelected,
+    refetchCurrentConversation,
+    setCurrentConversation,
+    setCurrentSystemPromptId,
+  } = useCurrentConversation({
     allSystemPrompts,
-    areConnectorsFetched,
-    conversationTitle,
     conversations,
-    conversationsLoaded,
-    currentConversationId,
-    getDefaultConversation,
-    getLastConversationId,
-    isAssistantEnabled,
+    defaultConnector,
+    refetchCurrentUserConversations,
+    conversationId: getLastConversationId(conversationTitle),
+    mayUpdateConversations:
+      isFetchedConnectors &&
+      isFetchedCurrentUserConversations &&
+      isFetchedPrompts &&
+      Object.keys(conversations).length > 0,
+  });
+
+  const isInitialLoad = useMemo(() => {
```

Comment:
@spong I'm considering using the `AssistantAnimatedIcon` for the assistant body loading view, but since it is not available in Eui yet, I cannot get that same bounce effect from `EuiLoadingLogo`. Which do you prefer?
```
// proper loading component 
<EuiEmptyPrompt icon={<EuiLoadingLogo logo="logoSecurity" size="xl" />} />

// option with AssistantAnimatedIcon
<EuiEmptyPrompt icon={<AssistantAnimatedIcon />} />
```

https://github.com/user-attachments/assets/73b1a79a-8a5e-4d83-97b7-01a8f012b08a

Reply:
Might be best for us to confer with @bojanasan & design here since they're handling all the alignment stuff. This is just on initial load as conversations/prompts and all that get set up, right? If we don't have to gate the full UI with a loader I think that would be best UX (and less jitter), but if not what you showed in option2 above looks good to me.

Reply:
Ok i sent this thread to @bojanasan

Reply:
Re: jitter. Without a loading option, there are 3 views shown before showing the proper view:
<img width="1440" alt="Screenshot 2024-08-14 at 8 47 07 AM" src="https://github.com/user-attachments/assets/f8989f04-78ae-4a6e-ba54-f11a7bc4c18f">
<img width="1438" alt="Screenshot 2024-08-14 at 8 48 26 AM" src="https://github.com/user-attachments/assets/9ca43987-aec5-4ea0-827b-f639dd35b2a7">
<img width="1440" alt="Screenshot 2024-08-14 at 8 46 52 AM" src="https://github.com/user-attachments/assets/a4029188-211e-4903-9f31-a6debb2f4bab">
<img width="721" alt="Screenshot 2024-08-14 at 8 47 00 AM" src="https://github.com/user-attachments/assets/8ba1f4f2-b5a1-473c-825e-8d1f006ee0e0">

With the loading UI, there are 2 views: loading and the proper view. No confusion as to if what you're seeing is correct

Reply:
@stephmilovic I think the loading state with the AI logo works well. I think it looks smoother than the other option, and its nice to use the existing AI icon animation. 

Quick question, it would be great if the body content was centered vertically. Is it off right now because of the browser page height?

Reply:
https://github.com/elastic/kibana/pull/190151/commits/c1edeec91ac48b57e27599cb5926d8e6fde8a9dc